### ISSUE TITLE
fixed dependency versions hurt my soul

### ISSUE THREAD

**Issue by brandondube** [Jashcraf/poke#89]
I created a conda environment for poke:
```
$ conda create -n poke python=3.11 msgpack-python jupyterlab numpy scipy matplotlib astropy
```

Then upon pip install, it clobbered a lot of those with pinned versions:

<details><summary><code>$ pip install -e .</code></summary><br><pre>
Obtaining file:///Users/bdube/src/poke
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Requirement already satisfied: numpy>=1.23.5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from poke==1.0) (1.25.2)
Requirement already satisfied: matplotlib>=3.6.2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from poke==1.0) (3.7.1)
Collecting astropy>=5.1.1 (from poke==1.0)
  Obtaining dependency information for astropy>=5.1.1 from https://files.pythonhosted.org/packages/5a/f7/c7030f9d1190bcc49564432a4cb536494aecd7d3fcb654cc5d604c802607/astropy-5.3.2-cp311-cp311-macosx_11_0_arm64.whl.metadata
  Downloading astropy-5.3.2-cp311-cp311-macosx_11_0_arm64.whl.metadata (9.4 kB)
Collecting zosapi==0.2 (from poke==1.0)
  Downloading zosapi-0.2.tar.gz (3.4 kB)
  Preparing metadata (setup.py) ... done
Collecting numexpr>=2.8.4 (from poke==1.0)
  Obtaining dependency information for numexpr>=2.8.4 from https://files.pythonhosted.org/packages/00/7b/e9cbb3a7fd7baa3fadfa1e710fdd6be939d106d25058b43ea49da1ec6229/numexpr-2.8.5-cp311-cp311-macosx_11_0_arm64.whl.metadata
  Downloading numexpr-2.8.5-cp311-cp311-macosx_11_0_arm64.whl.metadata (8.0 kB)
Collecting msgpack>=1.0.5 (from poke==1.0)
  Downloading msgpack-1.0.5-cp311-cp311-macosx_11_0_arm64.whl (69 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 69.0/69.0 kB 4.9 MB/s eta 0:00:00
Collecting msgpack-numpy>=0.4.8 (from poke==1.0)
  Downloading msgpack_numpy-0.4.8-py2.py3-none-any.whl (6.9 kB)
Collecting numpydoc==1.5.0 (from poke==1.0)
  Downloading numpydoc-1.5.0-py3-none-any.whl (52 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 52.4/52.4 kB 4.9 MB/s eta 0:00:00
Collecting nbsphinx==0.8.11 (from poke==1.0)
  Downloading nbsphinx-0.8.11-py3-none-any.whl (26 kB)
Collecting docutils (from nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for docutils from https://files.pythonhosted.org/packages/26/87/f238c0670b94533ac0353a4e2a1a771a0cc73277b88bff23d3ae35a256c1/docutils-0.20.1-py3-none-any.whl.metadata
  Downloading docutils-0.20.1-py3-none-any.whl.metadata (2.8 kB)
Requirement already satisfied: jinja2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbsphinx==0.8.11->poke==1.0) (3.1.2)
Requirement already satisfied: nbconvert!=5.4 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbsphinx==0.8.11->poke==1.0) (6.5.4)
Requirement already satisfied: traitlets>=5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbsphinx==0.8.11->poke==1.0) (5.7.1)
Requirement already satisfied: nbformat in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbsphinx==0.8.11->poke==1.0) (5.7.0)
Collecting sphinx>=1.8 (from nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinx>=1.8 from https://files.pythonhosted.org/packages/40/c0/e62ce9d243bfa2d9f290d7c730c7df3d9e979f0016e38c767f33ffcc2185/sphinx-7.2.2-py3-none-any.whl.metadata
  Downloading sphinx-7.2.2-py3-none-any.whl.metadata (5.9 kB)
Collecting pythonnet (from zosapi==0.2->poke==1.0)
  Downloading pythonnet-3.0.1-py3-none-any.whl (284 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 284.5/284.5 kB 12.9 MB/s eta 0:00:00
Requirement already satisfied: pyerfa>=2.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from astropy>=5.1.1->poke==1.0) (2.0.0)
Requirement already satisfied: PyYAML>=3.13 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from astropy>=5.1.1->poke==1.0) (6.0)
Requirement already satisfied: packaging>=19.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from astropy>=5.1.1->poke==1.0) (23.0)
Requirement already satisfied: contourpy>=1.0.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (1.0.5)
Requirement already satisfied: cycler>=0.10 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (0.11.0)
Requirement already satisfied: fonttools>=4.22.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (4.25.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (1.4.4)
Requirement already satisfied: pillow>=6.2.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (9.4.0)
Requirement already satisfied: pyparsing>=2.3.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (3.0.9)
Requirement already satisfied: python-dateutil>=2.7 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from matplotlib>=3.6.2->poke==1.0) (2.8.2)
Requirement already satisfied: MarkupSafe>=2.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jinja2->nbsphinx==0.8.11->poke==1.0) (2.1.1)
Requirement already satisfied: lxml in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (4.9.2)
Requirement already satisfied: beautifulsoup4 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (4.12.2)
Requirement already satisfied: bleach in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (4.1.0)
Requirement already satisfied: defusedxml in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.7.1)
Requirement already satisfied: entrypoints>=0.2.2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.4)
Requirement already satisfied: jupyter-core>=4.7 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (5.3.0)
Requirement already satisfied: jupyterlab-pygments in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.1.2)
Requirement already satisfied: mistune<2,>=0.8.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.8.4)
Requirement already satisfied: nbclient>=0.5.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.5.13)
Requirement already satisfied: pandocfilters>=1.4.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (1.5.0)
Requirement already satisfied: pygments>=2.4.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (2.15.1)
Requirement already satisfied: tinycss2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (1.2.1)
Requirement already satisfied: fastjsonschema in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbformat->nbsphinx==0.8.11->poke==1.0) (2.16.2)
Requirement already satisfied: jsonschema>=2.6 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbformat->nbsphinx==0.8.11->poke==1.0) (4.17.3)
Requirement already satisfied: six>=1.5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from python-dateutil>=2.7->matplotlib>=3.6.2->poke==1.0) (1.16.0)
Collecting sphinxcontrib-applehelp (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinxcontrib-applehelp from https://files.pythonhosted.org/packages/c0/0c/261c0949083c0ac635853528bb0070c89e927841d4e533ba0b5563365c06/sphinxcontrib_applehelp-1.0.7-py3-none-any.whl.metadata
  Downloading sphinxcontrib_applehelp-1.0.7-py3-none-any.whl.metadata (2.2 kB)
Collecting sphinxcontrib-devhelp (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinxcontrib-devhelp from https://files.pythonhosted.org/packages/c0/03/010ac733ec7b7f71c1dc88e7115743ee466560d6d85373b56fb9916e4586/sphinxcontrib_devhelp-1.0.5-py3-none-any.whl.metadata
  Downloading sphinxcontrib_devhelp-1.0.5-py3-none-any.whl.metadata (2.2 kB)
Collecting sphinxcontrib-jsmath (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Downloading sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl (5.1 kB)
Collecting sphinxcontrib-htmlhelp>=2.0.0 (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinxcontrib-htmlhelp>=2.0.0 from https://files.pythonhosted.org/packages/28/7a/958f8e3e6abe8219d0d1f1224886de847ab227b218f4a07b61bc337f64be/sphinxcontrib_htmlhelp-2.0.4-py3-none-any.whl.metadata
  Downloading sphinxcontrib_htmlhelp-2.0.4-py3-none-any.whl.metadata (2.2 kB)
Collecting sphinxcontrib-serializinghtml>=1.1.5 (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinxcontrib-serializinghtml>=1.1.5 from https://files.pythonhosted.org/packages/95/d6/2e0bda62b2a808070ac922d21a950aa2cb5e4fcfb87e5ff5f86bc43a2201/sphinxcontrib_serializinghtml-1.1.9-py3-none-any.whl.metadata
  Downloading sphinxcontrib_serializinghtml-1.1.9-py3-none-any.whl.metadata (2.3 kB)
Collecting sphinxcontrib-qthelp (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Obtaining dependency information for sphinxcontrib-qthelp from https://files.pythonhosted.org/packages/1f/e5/1850f3f118e95581c1e30b57028ac979badee1eb29e70ee72b0241f5a185/sphinxcontrib_qthelp-1.0.6-py3-none-any.whl.metadata
  Downloading sphinxcontrib_qthelp-1.0.6-py3-none-any.whl.metadata (2.2 kB)
Collecting snowballstemmer>=2.0 (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Downloading snowballstemmer-2.2.0-py2.py3-none-any.whl (93 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 93.0/93.0 kB 11.1 MB/s eta 0:00:00
Requirement already satisfied: babel>=2.9 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (2.11.0)
Collecting alabaster<0.8,>=0.7 (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Downloading alabaster-0.7.13-py3-none-any.whl (13 kB)
Collecting imagesize>=1.3 (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0)
  Downloading imagesize-1.4.1-py2.py3-none-any.whl (8.8 kB)
Requirement already satisfied: requests>=2.25.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (2.31.0)
Collecting clr-loader<0.3.0,>=0.2.2 (from pythonnet->zosapi==0.2->poke==1.0)
  Obtaining dependency information for clr-loader<0.3.0,>=0.2.2 from https://files.pythonhosted.org/packages/b3/1a/2801c4e3888aac8fcd4d88d32a1155a778d1c01ca040ae8137d433a39cc6/clr_loader-0.2.6-py3-none-any.whl.metadata
  Downloading clr_loader-0.2.6-py3-none-any.whl.metadata (1.4 kB)
Requirement already satisfied: pytz>=2015.7 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from babel>=2.9->sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (2022.7)
Requirement already satisfied: cffi>=1.13 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from clr-loader<0.3.0,>=0.2.2->pythonnet->zosapi==0.2->poke==1.0) (1.15.1)
Requirement already satisfied: attrs>=17.4.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jsonschema>=2.6->nbformat->nbsphinx==0.8.11->poke==1.0) (22.1.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jsonschema>=2.6->nbformat->nbsphinx==0.8.11->poke==1.0) (0.18.0)
Requirement already satisfied: platformdirs>=2.5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jupyter-core>=4.7->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (3.10.0)
Requirement already satisfied: jupyter-client>=6.1.5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbclient>=0.5.0->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (7.4.9)
Requirement already satisfied: nest-asyncio in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from nbclient>=0.5.0->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (1.5.6)
Requirement already satisfied: charset-normalizer<4,>=2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from requests>=2.25.0->sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from requests>=2.25.0->sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (3.4)
Requirement already satisfied: urllib3<3,>=1.21.1 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from requests>=2.25.0->sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (1.26.16)
Requirement already satisfied: certifi>=2017.4.17 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from requests>=2.25.0->sphinx>=1.8->nbsphinx==0.8.11->poke==1.0) (2023.7.22)
Requirement already satisfied: soupsieve>1.2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from beautifulsoup4->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (2.4)
Requirement already satisfied: webencodings in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from bleach->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (0.5.1)
Requirement already satisfied: pycparser in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from cffi>=1.13->clr-loader<0.3.0,>=0.2.2->pythonnet->zosapi==0.2->poke==1.0) (2.21)
Requirement already satisfied: pyzmq>=23.0 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jupyter-client>=6.1.5->nbclient>=0.5.0->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (23.2.0)
Requirement already satisfied: tornado>=6.2 in /Users/bdube/miniconda3/envs/poke/lib/python3.11/site-packages (from jupyter-client>=6.1.5->nbclient>=0.5.0->nbconvert!=5.4->nbsphinx==0.8.11->poke==1.0) (6.3.2)
Downloading astropy-5.3.2-cp311-cp311-macosx_11_0_arm64.whl (7.0 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.0/7.0 MB 21.9 MB/s eta 0:00:00
Downloading numexpr-2.8.5-cp311-cp311-macosx_11_0_arm64.whl (90 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 90.9/90.9 kB 9.4 MB/s eta 0:00:00
Downloading sphinx-7.2.2-py3-none-any.whl (3.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.2/3.2 MB 20.7 MB/s eta 0:00:00
Downloading docutils-0.20.1-py3-none-any.whl (572 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 572.7/572.7 kB 16.3 MB/s eta 0:00:00
Downloading clr_loader-0.2.6-py3-none-any.whl (51 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 51.3/51.3 kB 4.6 MB/s eta 0:00:00
Downloading sphinxcontrib_htmlhelp-2.0.4-py3-none-any.whl (99 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 99.2/99.2 kB 8.6 MB/s eta 0:00:00
Downloading sphinxcontrib_serializinghtml-1.1.9-py3-none-any.whl (92 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 92.7/92.7 kB 8.0 MB/s eta 0:00:00
Downloading sphinxcontrib_applehelp-1.0.7-py3-none-any.whl (120 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 120.0/120.0 kB 12.0 MB/s eta 0:00:00
Downloading sphinxcontrib_devhelp-1.0.5-py3-none-any.whl (83 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 83.5/83.5 kB 9.5 MB/s eta 0:00:00
Downloading sphinxcontrib_qthelp-1.0.6-py3-none-any.whl (89 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 89.4/89.4 kB 7.6 MB/s eta 0:00:00
Building wheels for collected packages: poke, zosapi
  Building editable for poke (pyproject.toml) ... done
  Created wheel for poke: filename=poke-1.0-0.editable-py3-none-any.whl size=3694 sha256=ca651d35d5a422a5418931019f48e46755a2ce4fa2eb3e9f26dae22097915a69
  Stored in directory: /private/var/folders/hw/82bzy1hn5zxdlfrksnvdmqf40000gq/T/pip-ephem-wheel-cache-f5wxa_ee/wheels/68/a6/2e/f1c1633df3038aa39d97de13ceb98190ef152a9c92747e56cb
  Building wheel for zosapi (setup.py) ... done
  Created wheel for zosapi: filename=zosapi-0.2-py3-none-any.whl size=3632 sha256=050ed499f6b325737f0f69d65546bce99f807138ee38b8ef1625e4622a7a438b
  Stored in directory: /Users/bdube/Library/Caches/pip/wheels/08/d2/50/ab3479d2ab1a3aad59b06ce65908c375c13f0205e46df2ff98
Successfully built poke zosapi
Installing collected packages: snowballstemmer, msgpack, sphinxcontrib-jsmath, numexpr, msgpack-numpy, imagesize, docutils, alabaster, clr-loader, astropy, pythonnet, zosapi, sphinxcontrib-serializinghtml, sphinxcontrib-qthelp, sphinxcontrib-htmlhelp, sphinxcontrib-devhelp, sphinxcontrib-applehelp, sphinx, numpydoc, nbsphinx, poke
  Attempting uninstall: msgpack
    Found existing installation: msgpack 1.0.3
    Uninstalling msgpack-1.0.3:
      Successfully uninstalled msgpack-1.0.3
  Attempting uninstall: astropy
    Found existing installation: astropy 5.1
    Uninstalling astropy-5.1:
      Successfully uninstalled astropy-5.1
Successfully installed alabaster-0.7.13 astropy-5.3.2 clr-loader-0.2.6 docutils-0.20.1 imagesize-1.4.1 msgpack-1.0.5 msgpack-numpy-0.4.8 nbsphinx-0.8.11 numexpr-2.8.5 numpydoc-1.5.0 poke-1.0 pythonnet-3.0.1 snowballstemmer-2.2.0 sphinx-7.2.2 sphinxcontrib-applehelp-1.0.7 sphinxcontrib-devhelp-1.0.5 sphinxcontrib-htmlhelp-2.0.4 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.6 sphinxcontrib-serializinghtml-1.1.9 zosapi-0.2

</pre></details>

It would be lovely if poke specified at worst lower bounds on versions, so that installing poke doesn't potentially change/break existing environments

**Comment by Jashcraf** [comment]
Roger that and can do! Would upper and lower bounds be better?

**Comment by brandondube** [comment]
upper bounds accelerate the heat death of the universe*, lower only please

\* mean that eventually your code can't be used with anything else unless you religiously update the dependency versions, which you probably won't experience any breaking changes from anyway (in ~10y, not pinning numpy and scipy versions has never hurt me)

**Comment by Jashcraf** [comment]
Mostly fixed a while ago, finished up everything in 884a2dd049468bf2d84a37381f17bb07d681ac55
